cpa/lint.go

```diff
@@ -0,0 +1,62 @@
+package cpa
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+)
+
+type LintRule func(*ast.Module) error
+
+func AllowedPackages(names ...string) LintRule {
+	return func(m *ast.Module) error {
+		packageName := strings.TrimPrefix(m.Package.String(), "package ")
+		for _, name := range names {
+			if name == packageName {
+				return nil
+			}
+		}
+		return lintErrf(
+			"invalid package name: expected one of packages [%s] but got %q",
+			strings.Join(names, ", "),
+			m.Package,
+		)
+	}
+}
+
+func DisallowMetaBranch() LintRule {
```

Comment:
We can make it more generic, like DisallowDeprecatedTerms()
Which takes a map of deprecated term to the replacement terms, if replacement term is empty, consider no recommended replacement.